docs/codeql/codeql-language-guides/customizing-library-models-for-csharp.rst

```diff
@@ -0,0 +1,331 @@
+.. _customizing-library-models-for-csharp:
+
+Customizing library models for C#
+=================================
+
+You can model the methods and callables that control data flow in any framework or library. This is especially useful for custom frameworks or niche libraries, that are not supported by the standard CodeQL libraries.
+
+.. include:: ../reusables/beta-note-customizing-library-models.rst
+
+About this article
+------------------
+
+This article contains reference material about how to define custom models for sources, sinks and flow summaries for C# dependencies in data extension files.
+
+About data extensions
+---------------------
+
+You can customize analysis by defining models (summaries, sinks, and sources) of your code's C#/.NET dependencies in data extension files. Each model defines the behavior of one or more elements of your library or framework, such as methods, properties, and callables. When you run dataflow analysis, these models expand the potential sources and sinks tracked by dataflow analysis and improve the precision of results.
```

Comment:
`customize analysis` -> `customize analyses`?

Reply:
And `dataflow analysis` -> `dataflow analyses`.

Reply:
Not sure which is better, but that wording is taken from the [Java customizing-library-models docs].

 [Java customizing-library-models docs]: https://github.com/github/codeql/blob/566351a49a24c00bca9352ad000775e99231b8ed/docs/codeql/codeql-language-guides/customizing-library-models-for-java-and-kotlin.rst?plain=1#L24